open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataEventConsumer.java

```diff
@@ -102,6 +103,21 @@ public void run()
         {
             try
             {
+            	//TBD - Does the kafka configuration need to be adjusted to prevent the consumer
+            	//from being declared "dead" if the polling does not occur often enough?  There is a Kafka
+            	//setting "max.poll.interval.ms" that may have some bearing on this.  The default value
+            	//of this setting is Integer.MAX_VALUE, so unless we are overriding that property
+            	//we *should* be ok.            	
+            	if (connector.getNumberOfUnprocessedEvents() > defaultMaxQueueSize) {
+            		//The connector queue is too big.  Wait until the size goes down until
+            		//polling again.  If we let the events just accumulate, we will
+            		//eventually run out of memory if the consumer cannot keep up.
+            		log.warn("Skipping Kafka polling since unprocessed message queue size {} is greater than {}", connector.getNumberOfUnprocessedEvents(), defaultMaxQueueSize);
+            		awaitNextPollingTime();

```

Comment:
There is kafka property 'max.poll.records'. This identifies the number of records which will be processed on each poll. By default in CAMS it was set to  500. We are reducing it to 100. So that will reduce the number of messages to come in. Essentially each poll will only give us 100 messages. Do we still want to block the thread? 
I think it is  un necessary.

As you said, if we are putting delays in code, then we need to be aware of session timing out as well.

Reply:
@guptaneeru Yes, this is needed.  We cannot allow the consumer to run out of memory.  The egeria code has no knowledge of what the kafka parameters are, and it needs to work in all kafka configurations.  In addition, tweaking kafka parameters does not eliminate the possibility of the event queue consuming all of the available heap memory.  This change does do that.

Reply:
FYI it appears that for egeria the defaults we end up using are:

        max.poll.interval.ms = 300000
        max.poll.records = 500

Reply:
Personally I'd be a little worried if the consumer didn't poll within 300s ... but note that we also have session.timeout.ms . This could be kept lower, whilst the max.poll.timeout.ms is increased - could be prudent configuration. However I agree with Jeff that there does need to be some limits here to prevent any java client running out of heap if message backlog gets very large (there could be a huge burst for example), so it is certainly better to leave many more of those in the Q.

I don't know the kafka connector well... but There may also be a reliability consideration here. If messages are read off the queue but buffered in the client there is a bigger exposure as the messages will likely not be read should the client/machine die & come up elsewhere, whilst if left in the topic they would?

In my current setup I see auto commit is true, and the auto commit interval is 1s - so reducing the max queue size (as per this PR) also reduces the exposure to lost events? I can imagine in some scenarios we may want them tightened up even further